src/ParallelKernel/allocators.jl

```diff
@@ -2,7 +2,13 @@
 const ZEROS_DOC = """
     @zeros(args...)
 
+!!! note "Advanced"
+        @zeros(numbertype, args...)
+
 Call `zeros(numbertype, args...)`, where `numbertype` is the datatype selected with [`@init_parallel_kernel`](@ref) and the function `zeros` is chosen to be compatible with the package for parallelization selected with [`@init_parallel_kernel`](@ref) (zeros for Threads and CUDA.zeros for CUDA).
+
+!!! note "Advanced"
+    The `numbertype` can be explicitly passed as argument in order to be used instead of the default `numbertype` chosen with [`@init_parallel_kernel`](@ref). If no default `numbertype` was chosen [`@init_parallel_kernel`](@ref), then the argument `numbertype` is mandatory. This needs to be used with care to ensure that no datatype conversions occurs in performance critical computations.
```

Comment:
no conversion occurs  -or-  no conversions occur

Reply:
Regarding the "advanced" doc, It is not fully clear if `numbertype` could be chosen differently for a single array initialisation then what is defined in `init_parallel_stencil` (if it is defined there), i.e., must all arrays have the same (custom) `numbertype` and has that `numbertype` to be the same as what is defined in `init_parallel_stencil`?

Reply:
It can be chosen differently from the default - I think it states that right in the docstring. `@zeros` etc will have to give an error if the chosen numbertype is not allowed. It should then be documented here as well what numbertypes are allowed and this will make it all clear. This is however left for the next PR, which will address exactly the usage of different numbertypes than (complex) float.